src/main/java/org/jabref/http/server/Server.java

```diff
@@ -37,26 +36,24 @@ public static void main(final String[] args) throws InterruptedException, URISyn
         SLF4JBridgeHandler.removeHandlersForRootLogger();
         SLF4JBridgeHandler.install();
 
-        final ObservableList<String> lastFilesOpened = JabRefPreferences.getInstance().getGuiPreferences().getLastFilesOpened();
+        final ObservableList<Path> lastFilesOpened = JabRefPreferences.getInstance().getGuiPreferences().getLastFilesOpened();
 
         // The server serves the last opened files (see org.jabref.http.server.LibraryResource.getLibraryPath)
         // In a testing environment, this might be difficult to handle
         // This is a quick solution. The architectural fine solution would use some http context or other @Inject_ed variables in org.jabref.http.server.LibraryResource
         if (args.length > 0) {
             LOGGER.debug("Command line parameters passed");
-            List<String> filesToAdd = Arrays.stream(args)
-                                            .map(Path::of)
-                                            .filter(Files::exists)
-                                            .map(Path::toString)
-                                            .filter(path -> !lastFilesOpened.contains(path))
-                                            .toList();
+            List<Path> filesToAdd = Arrays.stream(args)
+                                          .map(Path::of)
+                                          .filter(Files::exists)
+                                          .filter(path -> !lastFilesOpened.contains(path))
+                                          .toList();
 
             LOGGER.debug("Adding following files to the list of opened libraries: {}", filesToAdd);
 
             // add the files in the front of the last opened libraries
-            Collections.reverse(filesToAdd);
-            for (String path : filesToAdd) {
-                lastFilesOpened.add(0, path);
+            for (Path path : filesToAdd.reversed()) {
+                lastFilesOpened.addFirst(path);
```

Comment:
We reverse the list to add the content reverse?

Doesn't `lastfilesOpened.addAll(filesToAdd)` work?

Reply:
You tell me, you implemented it the first time. 😅

Reply:
Actually yes, read it again. The files to add should be added at the beginning, but in its correct order. addFirst or add(0,...) can only add one element, so by reversing it, you start with the last one of filesToAdd, put it at the beginning of lastFilesOpened. Then you take the second last, and put it also at the beginning, so the last one of filesToAdd is now the second in lastFilesOpened.

You actually thought this through when you wrote it. 😉